Temperature_API/router.py

```diff
@@ -0,0 +1,46 @@
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from . import schemas, crud
+from dependencies import get_db
+from .schemas import TemperatureList
+
+router = APIRouter()
+
+
+# Get all temperature records
+@router.get("/temperatures/", response_model=list[schemas.TemperatureList])
+async def get_all_temperatures(db: AsyncSession = Depends(get_db)):
+    """
+    Get a list of all temperature records.
+    """
+    return await crud.get_all_temperatures(db=db)
+
+
+# Get temperatures for city by id
+@router.get(
+    "/temperatures/{city_id}/",
+    response_model=TemperatureList
```

Comment:
Ensure that the `TemperatureList` schema is correctly defined and imported. The response model should match the expected output format for the endpoint.